### ISSUE TITLE
largestFreeBlockInternal at boot is ~13KB lower than fb943db's — same idle pthread/task-creation cliff, reached much faster

### ISSUE THREAD

**Issue by GlomarGadaffi** [GlomarGadaffi/pocket-dial#366]
## Summary

Attempting #328's zero-alloc proof on a freshly-flashed `.244` (`main@db08853`, plain non-`heap_trace` shipping build), I never got to place the scripted call — the board hit `pthread: Failed to create task!` / `[HttpServer] connection thread spawn failed: Not enough space` at uptime ~1808s (30 min), doing nothing but idle SIP re-registration. A clean re-run isolated the likely real story: **`largestFreeBlockInternal` at t≈120s is ~13 KB lower on this build than the last comparable measurement on `fb943db`** (14324 vs 27628) — a much shorter runway before the same pre-existing erosion mechanism reaches the cliff, not evidence that #278's fix regressed.

## First run: signature counts (confounded by my own polling — see caveat)

7× `pthread: Failed to create task!`, paired 1:1 with 7× `connection thread spawn failed`. **Zero** `spicommon_dma_setup_priv_buffer... Failed to allocate priv TX buffer` (#328's own signature), zero `xTaskCreatePinnedToCore failed`, zero `esp-aes: Failed to allocate`. SIP re-registration kept working the whole time — this is HTTP-connection-thread creation specifically, the #278 shape, not #328's RTP-DMA-churn shape.

Caveat: my own `/api/status` polling (~1 req/min) plus a couple of manual curls were the only HTTP traffic in that window — not a clean idle baseline. Board was healthy and serving JSON at uptime 297s right after flashing.

## Clean control-arm run: single poller, nothing else touching the board

Rebooted `.244` clean. One purpose-built poller hitting `/api/status` every 60s, no other traffic, no manual curls:

```
t=0s    freeInt=37111  minInt=10616  largestInt=14324  freeDma=28791  largestDma=14324
t=60s   freeInt=38139  minInt=10616  largestInt=14324  freeDma=29803  largestDma=14324
t=120s  freeInt=36023  minInt=10616  largestInt=14324  freeDma=27687  largestDma=14324
t=180s  freeInt=34431  minInt=10616  largestInt=14324  freeDma=26095  largestDma=14324
t=240s  freeInt=33307  minInt=10616  largestInt=14324  freeDma=24971  largestDma=14324
t=300s  freeInt=31903  minInt=10616  largestInt=14324  freeDma=23567  largestDma=14324
t=360s  freeInt=30919  minInt=10616  largestInt=10228  freeDma=22583  largestDma=10228
t=420s  freeInt=30255  minInt=10616  largestInt=10228  freeDma=21919  largestDma=10228
t=480s  freeInt=30251  minInt=10616  largestInt=10228  freeDma=21915  largestDma=10228
t=540s  freeInt=28979  minInt=10616  largestInt=10228  freeDma=20643  largestDma=10228
t=600s  freeInt=28343  minInt=10616  largestInt=10228  freeDma=20007  largestDma=10228
t=660s  freeInt=27731  minInt=10616  largestInt=9204   freeDma=19395  largestDma=9204
t=720s  freeInt=27727  minInt=10616  largestInt=9204   freeDma=19391  largestDma=9204
t=780s  freeInt=27735  minInt=10616  largestInt=9204   freeDma=19399  largestDma=9204
```

Notes on reading this:
- `freeHeapInternal`/`freeHeapDma` decline fairly steadily from t=0 to ~t=600 (~800-900 B/min), then flatten from t=660-780 (barely moving). Not necessarily monotonic forever — this may be phased rather than a constant-rate leak. Data collection is ongoing past this point.
- `largestFreeBlockInternal`/`largestFreeBlockDma` step down in discrete jumps rather than eroding continuously: 14324 → 10228 (between t=300-360, ~4 KB) → 9204 (between t=600-660, ~1 KB), flat otherwise. **Neither step correlates with any log line** — I checked the serial capture for both windows and there's nothing but the routine 30s heartbeat/re-register pairs, no `TelephonyAnchor`, `HoldMusic`, SD, or error lines nearby. Silent, not attributable to a specific logged event.
- `minFreeHeapInternal` fixed at 10616 the entire run — that low-water mark was set before my first sample (my poller's first request lands ~30s after the hard-reset, after ETH bring-up and whatever boot-time allocations already ran), not a literal t=0 reading.
- Sanity check this is really `db08853`: heartbeat carries `ResetReason: USB` (#340's field, absent on any `fb943db`-family boot all last night).

## The actual lead: less runway at boot, not a new leak

ModestMouse's `fb943db` heap_trace run (2026-09-20, posted on #328) measured `largestFreeBlockInternal` at t=120s: **27628**. This run, same metric, same board, t=120s: **14324**. That's ~13 KB less headroom before any erosion has had time to matter, on a plain build vs. a heap_trace build (not perfectly comparable, but heap_trace should only cost *more* overhead, not give more headroom — so the gap is likely conservative if anything). If #278's fix is still doing its job (the pcap ring), a fixed erosion rate hitting a runway that's roughly half as long would explain reaching the same `pthread` cliff around 30 min instead of the ~90 min #278's own writeup measured.

**Ruled out:** microSD mounting as the new cost. `git log -S 'sd_mount()' -- main/esp_main_eth.cpp` shows it landed in `cfca63d`/`04dad59`, both confirmed ancestors of `fb943db` (`git merge-base --is-ancestor` true) — SD has been on this board the whole time, not a new variable.

**Checked, not fully ruled out:** `TrunkResolver`. `RequestsHandler::tick()` calls `_sipTrunk.sweep()` and (when `_sipTrunk.config().valid()`) `_trunkResolver.resolve()` every ~1s — this board has a leftover trunk dialplan row from earlier testing tonight. Zero `TrunkResolver` log lines in either capture rules out **repeated DNS-worker spawning** specifically (`resolve()` short-circuits on `Hit`/`Pending`/`Failed` without spawning a task, so silence is exactly what a cache hit looks like too — this isn't proof the trunk is idle-cost-free). I have not examined the rest of `_sipTrunk.sweep()`'s per-tick work.

**Not checked at all:** where the ~13 KB gap actually comes from. The commit range between `fb943db` and `db08853` spans essentially all of tonight's and last night's work (#310, #315/#319, #318, #330, #164 slices, #349-354, #359/#360, and more) — too wide to guess at. A bisect (or just reading each PR's stated worst-case RAM/stack cost) would find it; I haven't done that.

## Cross-refs

- #278 (closed) — same outward symptom (idle → `pthread`/task-creation failure). This isn't evidence #278's fix regressed; it looks like the same residual mechanism reaching the cliff faster because there's less runway now.
- #328 (open) — still blocked. Its own signature (`priv TX buffer`) appeared zero times in either capture. The MoH-based zero-alloc test (HoldMusic → `l2rtp::EgressChannel` → `DmaFramePool`) is written, syntax-checked, and posted on #294 — needs the board to stay healthy through a 900s window to run.
- #330 (merged) — not implicated by this data.

## Repro

`.244`, `main@db08853`, plain `idf.py -D SIP_TRANSPORT=eth build`, no calls placed, real T29 on ext 1001 registered and re-registering normally throughout. Raw serial capture and poll log are on glolab if anyone wants to look at the raw data before I post more.

**Comment by GlomarGadaffi** [comment]
## Root cause: it isn't a leak, and the threshold is 8192

Kept the control arm running to t=1320s. Two corrections to what I wrote above, then the actual cause.

**Correction 1 — nothing is leaking.** `freeHeapInternal` went flat and stayed there:

```
t=660s   27731      t=1020s  27731
t=720s   27727      t=1080s  27743
t=780s   27735      t=1140s  27763
t=840s   27727      t=1200s  27739
t=900s   27711      t=1260s  28055
```

Twenty minutes, ~±50 bytes of noise, no downward trend. The decline I reported from t=0-600 is a settling curve, not a leak. Whatever I call this, "leak" is the wrong word and the issue title's framing should follow.

**Correction 2 — drop the "300s period" and the ratchet.** I claimed `largestFreeBlockInternal` stepped down on a 300s clock. The next sample broke it: t=1260s read 7924, t=1320s read **9204** — it went back *up*. It oscillates (7924 ↔ 9204 in the later window), it does not ratchet. Three points was not a period and I should not have called it one.

**The actual cause.** `sdkconfig.defaults:108` sets `CONFIG_PTHREAD_TASK_STACK_SIZE_DEFAULT=8192`, and `HttpServer::acceptLoop()` spawns one detached `std::thread` per connection:

```cpp
std::thread([this, clientSock]() {
    handleClient(clientSock);
}).detach();
```

So **every HTTP request needs one contiguous 8192-byte internal-DRAM block** for its stack. The largest block available is oscillating from 7924 to 9204 — straddling that number. Above it, the request is served. Below it, `pthread: Failed to create task!` and the connection is dropped by the `catch` two lines down.

That accounts for every part of the symptom that previously looked odd:
- **Why intermittent rather than dead** — the reading crosses 8192 in both directions.
- **Why bursts of 7** — a dip lasts across several requests, then recovers.
- **Why SIP was unaffected** — the SIP tasks were created at boot out of an unfragmented heap and never re-allocate a stack.
- **Why it reads as "30 minutes of idle"** — it's not a duration at all, it's fragmentation settling to a level that happens to sit on top of the threshold.

There is no unbounded connection limit either, so N concurrent requests want N × 8 KB of internal DRAM — a separate robustness problem worth its own issue, not this one.

## Fix in progress

`esp_pthread_set_cfg()` at the top of `acceptLoop()` to give the connection threads a 4096-byte stack, leaving `CONFIG_PTHREAD_TASK_STACK_SIZE_DEFAULT` alone (it's every `std::thread` in the firmware, and nothing here measured the others). 4096 is supported by what runs on that thread: no TLS anywhere in HttpServer (zero `mbedtls`/`esp_tls` references), no large stack arrays declared, responses built through heap-backed `std::string`/`ostringstream`, and the multi-MB OTA/MoH uploads explicitly streamed in chunks rather than buffered.

Same PR adds `stackHwm_http_conn` to `/api/status` — the worst free-stack figure any connection thread has reported since boot — so the 4096 stops being my estimate and becomes a measured number anyone can check.

Hardware pass criterion: zero `pthread` lines across a window where `largestFreeBlockInternal` dips below 8192. That is also the stable-board precondition #328's MoH zero-alloc test has been blocked on.

**Still open and NOT addressed by this fix:** what fragments internal DRAM down to this level in the first place, and the ~13 KB largest-block gap versus `fb943db` described above. This fix moves the requirement out of the danger zone; it does not defragment anything. If the underlying floor keeps dropping, 4096 buys headroom, not immunity.

**Comment by GlomarGadaffi** [comment]
## Withdrawing the "~13 KB less runway" framing in the title and body

Fix is up as #367 and hardware-verified, but the headline comparison in this issue does not hold and I'd rather retract it than leave it for someone to build on.

I compared `largestFreeBlockInternal` at "t=120s" against ModestMouse's `fb943db` figure of 27628 at t=120s. Two independent problems:

1. **Different build profiles.** Theirs was a `heap_trace` build, mine a plain one.
2. **Different points on the settling curve.** My `t=` values are **poller-relative, not uptime-relative**, and the two runs started at very different offsets — the pre-fix run's `t=0` was around **uptime 445s**, the post-fix run's `t=0` at **uptime 29s**. So "t=120s" meant uptime ~565s in one and ~149s in the other.

On top of both: `/api/status` is served *on* a connection thread, so every reading was taken with that thread's 8192-byte stack already outstanding. The gauge consumes the block it measures, which depresses the very number being compared.

So: no conclusion about a runway gap versus `fb943db` should be carried forward from this issue. The **mechanism** finding stands unaffected and is what #367 fixes — 8192-byte contiguous requirement per connection vs. a largest hole oscillating 7924↔9204. An uptime-aligned run is collecting a figure that can actually be compared.

Retitling suggestion for whoever has the bit: something like "HTTP connection threads need a contiguous 8192B stack and fail when fragmentation puts the largest hole below it" describes what was actually found.
